### PR TITLE
Upgrade to dotnet SDK 8.0.303

### PR DESCRIPTION
**Pull request by m-ringler**
## Description

* Upgrade czishrink to dotnet SDK 8.0.303
* Disable megalinter for C# source files (the tools aren't working properly, and their rules partially conflict with StyleCop.Analyzers)
* Fix various megalinter warnings and errors (in particular YAML schema validation with V8R)
* Add a configuration file for YAML schema validation in vscode (necessary because V8R produces unhelpful error messages like './.mega-linter.yml# must NOT have additional properties' that don't really tell you what's wrong)

With this PR all github actions should be up and running again.

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I followed the Contributing Guidelines.
- [x] I did a self-review.
- [x] I updated the version of czicompress 
- [x] New and existing unit tests pass locally with my changes.

### CODE REVIEW

**Review by ptahmose — APPROVED** [review]
I assume the ".vscode"-files are added on purpose? Or - you explained it in the description, so I assume this is correct.

LGTM, thanks!

File: .vscode/settings.json
Comment (DaveyJonesBitPail) [inline .vscode/settings.json:1]: Are we using these schemas in the CI linting? Or were they just for local testing of lints?

**Review by DaveyJonesBitPail — APPROVED** [review]
Looks good to me! Thank you very much :)

File: .vscode/settings.json
Comment (DaveyJonesBitPail) [reply .vscode/settings.json:1]: Ah, you answered this in the description already

**Pull request merged**